depends/toolchain.cmake.in

```diff
@@ -0,0 +1,135 @@
+  split_compiler_launcher(DEPENDS_CXX_COMPILER_WITH_LAUNCHER CMAKE_CXX_COMPILER_LAUNCHER CMAKE_CXX_COMPILER)
+  if(CMAKE_VERSION VERSION_GREATER_EQUAL 3.21)
+    set(CMAKE_CXX_LINKER_LAUNCHER ${CMAKE_CXX_COMPILER_LAUNCHER})
+  endif()
+  if(CMAKE_VERSION VERSION_LESS 3.19)
+    set(DEPENDS_CXX_COMPILER_FLAGS ${CMAKE_CXX_COMPILER})
+    list(REMOVE_AT DEPENDS_CXX_COMPILER_FLAGS 0)
+    string(REPLACE ";" " " DEPENDS_CXX_COMPILER_FLAGS "${DEPENDS_CXX_COMPILER_FLAGS}")
+    list(GET CMAKE_CXX_COMPILER 0 CMAKE_CXX_COMPILER)
+  endif()
+endif()
+set(CMAKE_CXX_FLAGS_INIT "${DEPENDS_CXX_COMPILER_FLAGS} @CPPFLAGS@ @CXXFLAGS@")
+
+if(NOT CMAKE_OBJCXX_COMPILER)
+  set(DEPENDS_OBJCXX_COMPILER_WITH_LAUNCHER @CXX@)
+  split_compiler_launcher(DEPENDS_OBJCXX_COMPILER_WITH_LAUNCHER CMAKE_OBJCXX_COMPILER_LAUNCHER CMAKE_OBJCXX_COMPILER)
+  if(CMAKE_VERSION VERSION_GREATER_EQUAL 3.21)
+    set(CMAKE_OBJCXX_LINKER_LAUNCHER ${CMAKE_OBJCXX_COMPILER_LAUNCHER})
+  endif()
+  if(CMAKE_VERSION VERSION_LESS 3.19)
+    set(DEPENDS_OBJCXX_COMPILER_FLAGS ${CMAKE_OBJCXX_COMPILER})
+    list(REMOVE_AT DEPENDS_OBJCXX_COMPILER_FLAGS 0)
+    string(REPLACE ";" " " DEPENDS_OBJCXX_COMPILER_FLAGS "${DEPENDS_OBJCXX_COMPILER_FLAGS}")
+    list(GET CMAKE_OBJCXX_COMPILER 0 CMAKE_OBJCXX_COMPILER)
+  endif()
+endif()
+set(CMAKE_OBJCXX_FLAGS_INIT "${DEPENDS_OBJCXX_COMPILER_FLAGS} @CPPFLAGS@ @CXXFLAGS@")
+
+set(CMAKE_AR "@AR@")
+set(CMAKE_RANLIB "@RANLIB@")
+set(CMAKE_STRIP "@STRIP@")
+set(CMAKE_OBJCOPY "@OBJCOPY@")
+set(CMAKE_INSTALL_NAME_TOOL "@INSTALL_NAME_TOOL@")
+set(OTOOL "@OTOOL@")
+
+set(CMAKE_FIND_ROOT_PATH "@depends_prefix@")
+set(CMAKE_FIND_ROOT_PATH_MODE_PROGRAM NEVER)
+set(CMAKE_FIND_ROOT_PATH_MODE_LIBRARY ONLY)
+set(CMAKE_FIND_ROOT_PATH_MODE_INCLUDE ONLY)
+set(CMAKE_FIND_ROOT_PATH_MODE_PACKAGE ONLY)
+set(PKG_CONFIG_PATH "@depends_prefix@/lib/pkgconfig")
+if("@allow_host_packages@" STREQUAL "1")
+  set(DEPENDS_ALLOW_HOST_PACKAGES TRUE)
+else()
+  set(DEPENDS_ALLOW_HOST_PACKAGES FALSE)
+  set(PKG_CONFIG_LIBDIR "${PKG_CONFIG_PATH}")
+endif()
+set(QT_TRANSLATIONS_DIR "@depends_prefix@/translations")
+
+if(NOT WITH_GUI AND "@no_qt@" STREQUAL "1")
+  set(WITH_GUI "no" CACHE STRING "")
+endif()
+
+if(NOT WITH_QRENCODE AND "@no_qr@" STREQUAL "1")
+  set(WITH_QRENCODE OFF CACHE STRING "")
+endif()
+
+if(NOT WITH_ZMQ AND "@no_zmq@" STREQUAL "1")
+  set(WITH_ZMQ OFF CACHE STRING "")
+endif()
+
+if(NOT ENABLE_WALLET AND "@no_wallet@" STREQUAL "1")
+  set(ENABLE_WALLET OFF CACHE BOOL "")
+endif()
+
+if(NOT WITH_BDB AND "@no_bdb@" STREQUAL "1")
+  set(WITH_BDB OFF CACHE STRING "")
+endif()
+
+if(NOT WITH_SQLITE AND "@no_sqlite@" STREQUAL "1")
+  set(WITH_SQLITE OFF CACHE STRING "")
+endif()
+
+if(NOT WITH_MINIUPNPC AND "@no_upnp@" STREQUAL "1")
+  set(WITH_MINIUPNPC OFF CACHE STRING "")
+endif()
+
+if(NOT WITH_NATPMP AND "@no_natpmp@" STREQUAL "1")
+  set(WITH_NATPMP OFF CACHE STRING "")
+endif()
+
+if(NOT WITH_USDT AND "@no_usdt@" STREQUAL "1")
```

Comment:
Where is the line currently drawn between configuration already supported and configuration to be added later? We support usdt here, but don't support multiprocess, LTO, or some of the qt features.

Reply:
> Where is the line currently drawn between configuration already supported and configuration to be added later?

Well, it is vague, indeed.

> We support usdt here, but don't support multiprocess...

See: https://github.com/bitcoin/bitcoin/pull/25797/commits/1238ed54b09873b4249543d2f27c2e24051dcb66

> ... LTO...

Not implemented in the parent https://github.com/bitcoin/bitcoin/pull/25797 yet. Sorry about that.

> ... or some of the qt features.

It is more complicated. For example, see https://github.com/bitcoin/bitcoin/pull/25797/commits/50ac8b03a05564e146bc79ada504da5e950fa369.

Reply:
Thanks for the references, I'll be sure to come back to them later.